src/Adapter/Order/CommandHandler/IssuePartialRefundHandler.php

```diff
@@ -74,14 +84,19 @@ public function handle(IssuePartialRefundCommand $command)
         $amount = 0;
         $orderDetailList = [];
         $fullQuantityList = [];
+        $taxCalculator = $this->getTaxCalculator($order->carrier_tax_rate);
+        $translator = Context::getContext()->getTranslator();
 
         foreach ($refunds as $orderDetailId => $refund) {
-            $quantity = $refund['quantity'];
-
-            if (!$quantity) {
-                continue;
+            // this refund has an amount but no quantity, this should not happen
+            if (empty($refund['quantity'])) {
+                throw new OrderException(
+                    $translator->trans($this->quantityErrorMessage, array(), 'Admin.Orderscustomers.Notification')
```

Comment:
You mustn't use `$this->trans` with variable or the TranslationTool is unable to scan them and add them in the catalog Besides you shouldn't translate the exception message, the exception message is always in english and not translated 
You need to create two dedicated exceptions for these two use cases, for example `EmptyRefundQuantityException` and `EmptyRefundAmountException` then you will be able to translate these exceptions **in the controller** using `$this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));` and by adding a use case in `getErrorMessages` for each of your exceptions

Reply:
Besides getting the translator from the context is really ugly 😅